### PR TITLE
feat(mocha options): Add support for mocha options

### LINKED ISSUE #417: Is it possible to pass parameters to test runner?
In our project, mocha runs with options passed through file:
```
mocha --opts src/test/mocha.opts
```

Is there any way to pass this options when running stryker?

### PR DESCRIPTION
**Pull request by nicojs**
Add support for *some* mocha options to be passed to the `mocha` instance in the stryker-mocha-runner. The implementation uses a `ConfigEditor` to edit stryker config on the fly.

You can either pass a mocha.opts file using `mochaOptions: { opts: 'mocha.opts' }`, or pass values directly using `mochaOptions: { ui: 'qunit' }`.

Closes #417

### CODE REVIEW

File: packages/stryker-mocha-runner/src/MochaConfigEditor.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { ConfigEditor, Config } from 'stryker-api/config';
+import { getLogger } from 'log4js';
+import MochaRunnerOptions, { mochaOptionsKey } from './MochaRunnerOptions';
+
+
+export default class MochaConfigEditor implements ConfigEditor {
+
+  private log = getLogger('MochaTestRunner');
```
Comment (simondel) [inline packages/stryker-mocha-runner/src/MochaConfigEditor.ts]: Is this string intentional?

File: packages/stryker-mocha-runner/src/MochaConfigEditor.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  edit(config: Config): void {
+    const mochaOptions = config[mochaOptionsKey];
+    if (mochaOptions && mochaOptions.opts) {
```
Comment (simondel) [inline packages/stryker-mocha-runner/src/MochaConfigEditor.ts]: Shouldn't this be a `||` ?
The readme states:
> You can pass mocha options in 2 ways: **either** specify a mocha.opts file **or** configure `mochaOptions` in stryker.conf.js. **When both are specified**...

**Comment by simondel** [comment]
I have changed the name of the logger in the MochaConfigEditor because it contained the name of the MochaTestRunner

**Comment by nicojs** [comment]
Cool. That was a mistake indeed. The `&&` was intentional, however.

**Pull request merged**